core/contracts/financial-templates/implementation/ExpiringMultiParty.sol

```diff
@@ -5,5 +5,11 @@ import "./Liquidatable.sol";
 
 
 contract ExpiringMultiParty is Liquidatable {
-    constructor(ConstructorParams memory params) public Liquidatable(params) {}
+    constructor(ConstructorParams memory params)
+        public
+        Liquidatable(params)
+    // nonReentrant() This modifier is already applied on the FeePayer constructor.
```

Comment:
1. Shouldn't the guard be applied at the outer-most layer rather than the inner most? Otherwise, can't you have re-entrancy once you leave the inner-most call?
2. Does this fail if you add this here? Isn't the Liquidatable constructor called *before* the re-entrancy guard goes into effect?

Reply:
1. I think your counterexample is correct. I applied it on the innermost because I didn't want someone to construct a `PricelessPositionManager.sol` or `Liquidatable.sol` whose constructor wasn't guarded by `nonReentrant()`. In practice this might not be super useful since we are forcing everyone to deploy `EMP.sol`'s.
2. It doesn't actually fail if I add to `FeePayer` AND `EMP`, even though the bytecode does change (I verified).  I ended up adding to just the top-level one in `ExpiringMultiParty.sol`.